### PR TITLE
threat-model: Add missing threat-model document

### PR DESCRIPTION
**Pull request by amshinde**
This was added in the 1.x repo and is missing in the 2.x repo.
Copying over the document from 1.x.
For Kata 2.x, the block named as `kata-runtime` in the threat model
diagram has been changed to `kata-shim`.

This is a starting point and focuses on the devices / interfaces
with the virtual machine, and ultimately to the container itself.

We then discuss how these devices/interfaces vary by VMM/hypervisor.

The threat model drawing is created via gdocs, located here:
https://docs.google.com/drawings/d/1dPi9DG9bcCUXlayxrR2OUa1miEZXewtW7YCt4r_VDmA/edit?usp=sharing

Fixes: #2340

Signed-off-by: Eric Ernst <eric.ernst@intel.com>
Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @amshinde!

**Review by cmaf — APPROVED** [review]

**Comment by cmaf** [comment]
Hey @amshinde, static checks failed with:
```
ERROR: Document docs/threat-model/threat-model.md is not referenced
```

**Review by jodh-intel — APPROVED** [review]
Thanks @amshinde.

Let's get this landed then we can work on expanding it on a follow-up PR.

lgtm

**Comment by cmaf** [comment]
@amshinde static checks failed again: 

```
INFO: Spell checking file 'docs/threat-model/threat-model.md'
WARNING: Word 'sys': did you mean one of the following?: says, sis, syn, sysop
ERROR: Spell check failed for file: 'docs/threat-model/threat-model.md'
[static-checks.sh:824] INFO: spell check failed for document docs/threat-model/threat-model.md
[static-checks.sh:827] ERROR: spell check failed, See https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#spelling for more information.
make: *** [Makefile:34: static-checks] Error 1
```

File: docs/threat-model/threat-model.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+- Dynamic Resource Management: `ACPI` is utilized to allow for dynamic VM resource management (for example: CPU, memory, device hotplug). This is required when containers are resized,
+ or more generally when containers are added to a pod. 
+ 
+How these devices are utilized varies depending on the VMM utilized. We clarify the default settings provided when integrating Kata
+with the QEMU, Firecracker and Cloud Hypervisor VMMs in the following sections.
+
+### Devices
+
+Each virtio device is implemented by a backend, which may execute within userspace on the host (vhost-user), the VMM itself, or within the host kernel (vhost). While it may provide enhanced performance,
+vhost devices are often seen as higher risk since an exploit would be already running within the kernel space. While VMM and vhost-user are both in userspace on the host, `vhost-user` generally allows for the back-end process to require less sys-calls and capabilities compared to a full VMM.
```
Comment (jodh-intel) [inline docs/threat-model/threat-model.md:74]: You could change "sys-calls" to system calls to keep the spell checker happy.

**Pull request merged**